packages/@lwc/engine-core/src/framework/rendering.ts

```diff
@@ -625,15 +627,26 @@ function createViewModelHook(elm: HTMLElement, vnode: VCustomElement, renderer:
     return vm;
 }
 
-function allocateInSlot(vm: VM, children: VNodes) {
-    const { cmpSlots: oldSlots } = vm;
-    const cmpSlots = (vm.cmpSlots = create(null));
+/**
+ * Collects all slots into a SlotSet, traversing through VFragment Nodes
+ *
+ * @param vm
+ * @param children
+ * @param cmpSlots
+ */
+function collectSlots(vm: VM, children: VNodes, cmpSlots: SlotSet) {
```

Comment:
I've moved the collection logic into its own function that collects all slots into the same SlotSet. This way we don't drop sibling nodes when we collect them.